tests/test_read_lazy.py

```diff
@@ -0,0 +1,592 @@
+    file_name = f"orig_{worker_id}.{diskfmt}"
+    if diskfmt == "h5ad":
+        orig_path = tmp_path_factory.mktemp("h5ad_file_dir") / file_name
+    else:
+        orig_path = tmp_path_factory.mktemp(file_name)
+    orig = gen_adata((1000, 1100), mtx_format)
+    orig.raw = orig.copy()
+    getattr(orig, f"write_{diskfmt}")(orig_path)
+    return orig_path, orig
+
+
+@pytest.fixture
+def adata_remote_orig(
+    adata_remote_orig_with_path: tuple[Path, AnnData], load_annotation_index: bool
+) -> tuple[AnnData, AnnData]:
+    orig_path, orig = adata_remote_orig_with_path
+    return read_lazy(orig_path, load_annotation_index=load_annotation_index), orig
+
+
+@pytest.fixture(scope="session")
+def adata_remote_with_store_tall_skinny_path(
+    tmp_path_factory,
+    mtx_format,
+    worker_id: str = "serial",
+) -> Path:
+    orig_path = tmp_path_factory.mktemp(f"orig_{worker_id}.zarr")
+    M = 100_000  # forces zarr to chunk `obs` columns multiple ways - that way 1 access to `int64` below is actually only one access
+    N = 5
+    obs_names = pd.Index(f"cell{i}" for i in range(M))
+    var_names = pd.Index(f"gene{i}" for i in range(N))
+    obs = gen_typed_df(M, obs_names)
+    var = gen_typed_df(N, var_names)
+    orig = AnnData(
+        obs=obs,
+        var=var,
+        X=mtx_format(np.random.binomial(100, 0.005, (M, N)).astype(np.float32)),
+    )
+    orig.raw = orig.copy()
+    orig.write_zarr(orig_path)
+    return orig_path
+
+
+@pytest.fixture(scope="session")
+def adatas_paths_var_indices_for_concatenation(
+    tmp_path_factory, are_vars_different: bool, worker_id: str = "serial"
+) -> tuple[list[AnnData], list[Path], list[pd.Index]]:
+    adatas = []
+    var_indices = []
+    paths = []
+    M = 1000
+    N = 50
+    n_datasets = 3
+    for dataset_index in range(n_datasets):
+        orig_path = tmp_path_factory.mktemp(f"orig_{worker_id}_{dataset_index}.zarr")
+        paths.append(orig_path)
+        obs_names = pd.Index(f"cell_{dataset_index}_{i}" for i in range(M))
+        var_names = pd.Index(
+            f"gene_{i}{f'_{dataset_index}_ds' if are_vars_different and (i % 2) else ''}"
+            for i in range(N)
+        )
+        var_indices.append(var_names)
+        obs = gen_typed_df(M, obs_names)
+        var = gen_typed_df(N, var_names)
+        orig = AnnData(
+            obs=obs,
+            var=var,
+            X=np.random.binomial(100, 0.005, (M, N)).astype(np.float32),
+        )
+        orig.write_zarr(orig_path)
+        adatas.append(orig)
+    return adatas, paths, var_indices
+
+
+@pytest.fixture
+def concatenation_objects(
+    adatas_paths_var_indices_for_concatenation,
+) -> tuple[list[AnnData], list[pd.Index], list[AccessTrackingStore], list[AnnData]]:
+    adatas, paths, var_indices = adatas_paths_var_indices_for_concatenation
+    stores = [AccessTrackingStore(path) for path in paths]
+    lazys = [read_lazy(store) for store in stores]
+    return adatas, var_indices, stores, lazys
```

Comment:
please split this into 4 fixtures, that way typing below is much simpler, and you can just do

```py
def mystest(objs1, objs2, objs3, objs4): ...
```

instead of

```py
def mystest(concatenation_objects):
    objs1, objs2, objs3, objs4 = concatenation_objects
    ...
```